packages/adapter/src/chains/chainAdapter.ts

```diff
@@ -0,0 +1,144 @@
+  endpoint: string | undefined;
+  relayChain: string | undefined;
+  network: string | undefined;
+  paraId: number | undefined;
+  ss58Prefix: number | undefined;
+  name: string | undefined;
+  instructionWeight: Weight  | undefined;
+  xcmInstructionNetworkType: XcmInstructionNetworkType = XcmInstructionNetworkType.Null; 
+}
+
+export abstract class ChainAdapter {
+  api: ApiPromise | undefined;
+  protected chainData: ChainData;
+
+  /**
+   * Constructor
+   * @param api Polkadot API
+   * @param config Chain config
+   */
+  constructor(api: ApiPromise, config: ChainConfig) {
+    this.api = api;
+    this.chainData = new ChainData();
+    this.chainData.key = config.key;
+    this.chainData.assets = config.assets;
+    this.chainData.defaultAsset = config.defaultAsset;
+    this.chainData.instructionWeight = config.instructionWeight;
+    this.chainData.endpoint = config.endpoint;
+    this.chainData.network = config.network;
+    this.chainData.relayChain = config.relayChain;
+  }
+  
+  /**
+   * Initialize adapter
+   */
+  public abstract initialize(): Promise<void>;
+  
+  /**
+   * Calculate the derivative account ID of a certain account ID
+   * @param api Polkadot API
+   * @param accountId 
+   * @param paraId The paraId of the XCM message sender
+   * @param options Optional operation options: { locationType, network }
+   * @returns Derivative account
+   */
+  public abstract getDerivativeAccount(accountId: HexString, paraId: number, options?: any): HexString;
+
+  /**
+   * Calculate encoded call weight and overall weight for transact an extrinsic call through XCM message
+   * @param extrinsic The extrinsic that needs to be transacted
+   * @param account 
+   * @param instructionCount The number of XCM instructions
+   * returns { encodedCallWeight, overallWeight }
+   */
+  public abstract getXcmWeight(extrinsic: SubmittableExtrinsic<'promise'>, account: AddressOrPair, instructionCount: number): Promise<{ encodedCallWeight: Weight; overallWeight: Weight; }>;
+  
+  /**
+   * Calculate XCM execution fee based on weight
+   * @param weight 
+   * @param assetLocation 
+   * @returns XCM execution fee
+   */
+  public abstract weightToFee(weight: Weight, assetLocation: any): Promise<BN>;
+
+  /**
+   * Execute a cross-chain transfer
+   * @param destination The location of the destination chain
+   * @param recipient recipient account
+   * @param assetLocation Asset location
+   * @param assetAmount Asset amount
+   * @param keyringPair Operator's keyring pair
+   */
+  public abstract crossChainTransfer(destination: any, recipient: HexString, assetLocation: any, assetAmount: BN, keyringPair: KeyringPair): Promise<SendExtrinsicResult>;
+
+  /**
+   * Get polkadot API
+   * @returns Polkadot API
+   */
+  public getApi(): ApiPromise {
+    if (_.isUndefined(this.api)) throw new Error("Api not initialized");
+    return this.api;
+  }
+
+  /**
+   * Update chain data
+   */
+  public async updateChainData(): Promise<void> {
```

Comment:
Please call this function `getSs58AndParaId` which is basically what it does.

Reply:
The purpose of this function is to retrieve some basic data from the chain, such as paraId, ss58prefix, and so on. 

So, I think it shouldn't be renamed.

Reply:
If so, please do two things:
1. Change the function name to `fetchConfigsOnChain`, as in fetch usually means retrieving from a remote server.
2. Update the comment of the function;